src/transformers/modeling_utils.py

```diff
@@ -752,9 +755,15 @@ def load(module: nn.Module, prefix=""):
                 head_model_state_dict_without_base_prefix = [
                     key.split(cls.base_model_prefix + ".")[-1] for key in model.state_dict().keys()
                 ]
-
                 missing_keys.extend(head_model_state_dict_without_base_prefix - base_model_state_dict)
 
+            # Some models may have keys that are not in the state by design, removing them before needlessly warning
+            # the user.
+            if cls.keys_to_ignore_at_load is not None:
```

Comment:
(nit) you can init this on PretrainedModel to save an `if`. I like that better but I don't feel strongly.

Reply:
Changing a class variable when instantiating isn't very pythonic (or recommended), so I'd leave this as is.

Reply:
I meant setting it [here](https://github.com/huggingface/transformers/blob/edfd82f5ff179f7600d8f2eea204e21bd07d99e4/src/transformers/modeling_utils.py#L286-L285)